FormatFactory.py

```diff
@@ -1,27 +1,38 @@
 from GenotypeFormat import GenotypeFormat
 from GencallFormat import GencallFormat
+from BAlleleFreqFormat import BAlleleFreqFormat
+from LogRRatioFormat import LogRRatioFormat
+
 
 class FormatFactory(object):
     """FormatFactory is responsible for determining the fields
     that will be present in the FORMAT string and creating
     the appropriate corresponding format objects
     """
 
-    def __init__(self, no_samples, logger):
+    def __init__(self, no_samples, formats_to_include, logger):
```

Comment:
Should there be an error check if this includes a format for which there is no known handler?

Reply:
Yes good idea I'll add

Reply:
Added error checking in the class, and this is also checked during command line argument parsing